lib/spack/spack/spec.py

```diff
@@ -299,22 +301,29 @@ def _autospec(self, spec_like):
         return ArchSpec(spec_like)
 
     def _cmp_iter(self):
-        yield self.platform
+        yield self._platforms
         yield self.os
         yield self.target
 
     @property
     def platform(self):
         """The platform of the architecture."""
-        return self._platform
+        if len(self._platforms) == 1:
+            return self._platforms[0]
+        if len(self._platforms) == 0:
+            return None
+        raise ValueError(f"Platform not concrete: {self._platforms}")
 
     @platform.setter
-    def platform(self, value):
-        # The platform of the architecture spec will be verified as a
-        # supported Spack platform before it's set to ensure all specs
-        # refer to valid platforms.
-        value = str(value) if value is not None else None
-        self._platform = value
+    def platform(self, value: Union[spack.platforms.Platform, List[str], Optional[str]]):
+        if value is None:
+            self._platforms = ()
+        elif isinstance(value, str):
+            self._platforms = tuple(sorted(set(value.split(",")))) if "," in value else (value,)
```

Comment:
I realize it would raise the complexity of this method to a significant degree, but can we also support negation? Otherwise any operation specifying a variant/dep/etc for anywhere but one platform requires a user to be aware of all supported platforms rather than just the one they don't want.

If that's too much, that's very reasonable, and I'll just make this work with an "everywhere_but_windows" helper method that hardcodes the other platforms.

FWIW I'm more than willing to help with any extra development.

Reply:
Right now the platforms are just `linux`, (`cray`), `darwin`, `freebsd`, `windows`, so not too exhaustive and relatively easy to grep if we ever implement `not`

Reply:
I'd rather not do negation right now -- we will likely one day need it for everything, but I don't think today's the day.